expr/func_summarize_test.go

```diff
@@ -665,6 +665,78 @@ func TestSummarizeAlignToFrom(t *testing.T) {
 	testSummarize("AlignToFrom", input, outputMax[1], "45s", "max", true, t)
 }
 
+func TestSummarizeLargeIntervalTimestamps(t *testing.T) {
+
+	// This test is specifically testing the timestamp behavior, so we don't need real values.
+	// However, we do need a lot of datapoints to trigger the bug we are regression testing against.
+	interval1Day := uint32(24 * 60 * 60)
+	numTimestamps := uint32(2000)
```

Comment:
where does the 2000 come from? is the goal to test 2000 days?

Reply:
The bug in this case was two-fold. 

1. start-end caused an underflow for uint32
2. Division was by the interval, rather than the step (as it is in the python code) as well as using the adjusted start, end (rather than the original).

So, the bug comes about when (start-end) / interval is *less than* the number of input datapoints.

I needed a large enough number of input datapoints, and a large enough interval such that that `len(input) >  (start-end) / interval` (taking into account underflow). These were arbitrary numbers that I selected that caused the above to be true.